hw/display/virtio-dmabuf.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Virtio Shared dma-buf
+ *
+ * Copyright Red Hat, Inc. 2023
+ *
+ * Authors:
+ *     Albert Esteve <aesteve@redhat.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2 or later.
+ * See the COPYING file in the top-level directory.
+ */
+
+#include "hw/virtio/virtio-dmabuf.h"
+
+
+static struct shared_resources {
+    GHashTable *uuids;
+    bool locked;
+} res = {
+    .uuids = NULL,
+    .locked = false,
+};
+
+bool virtio_add_dmabuf(QemuUUID uuid, int udmabuf_fd)
+{
+    /*
+     * Check that the fd is valid.
+     */
+    return virtio_add_resource(uuid, GINT_TO_POINTER(udmabuf_fd));
+}
+
+bool virtio_add_resource(QemuUUID uuid, gpointer value)
+{
+    if (res.uuids == NULL) {
+        res.uuids = g_hash_table_new_full(NULL, NULL, NULL, g_free);
```

Comment:
Genuine question: Why does the method takes a `gpointer` when it expects an fd ? also, does it make sense to free the values with `g_free`?

Reply:
I see, thanks for the explanation!